internal/rule/rule.go

```diff
@@ -0,0 +1,33 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package rule
+
+import (
+	"log"
+)
+
+type Rule[T any] struct {
+	Conditions []ICondition[T]
+}
+
+func (r *Rule[T]) Evaluate(target T) (bool, error) {
```

Comment:
This looks like the validator pattern. I always felt the validator pattern is confusing and should only be used if there are many validations. If not, just have a couple of functions that execute one after the other.

Reply:
(I didn't even know there was a name to this but) What part is confusing, and how does that affect users of the interface? Why is having many validators make it better/different? Theoretically you can just keep adding separate functions that execute one after another no matter how many there are. Is it worth deleting interface that is already better prepped for more usecases and gives intuitive names to users?